yt_dlp/extractor/huya.py

```diff
@@ -0,0 +1,118 @@
+import base64
+import hashlib
+import json
+import random
+from html import unescape
+from urllib.parse import parse_qsl, urlencode
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    try_get
+)
+
+
+class HuyaIE(InfoExtractor):
+    _VALID_URL = r'(?x)^((?:http[s]?|fpt):)\/?\/(?:www\.|m\.|)huya\.com\/(?P<id>.*?)(?:\/|$)'
```

Comment:
```suggestion
    _VALID_URL = r'https?://(?:www\.|m\.)?huya\.com/(?P<id>[^/#?&])'
```
Keep regex clean.

(I'll partially withdraw this comment if you add tests for `fpt://` urls)